eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs

```diff
@@ -191,6 +201,38 @@ data BabbageTxBodyRaw era = BabbageTxBodyRaw
   }
   deriving (Generic, Typeable)
 
+-- We override this instance because the 'Sized' types also reference their
+-- serialisation and as such cannot be compared directly. An alternative would
+-- be to derive `EqRaw` for `Sized`.
+instance
+  (Era era, Eq (TxOut era), Eq (TxCert era), Eq (PParamsUpdate era)) =>
+  EqRaw (BabbageTxBodyRaw era)
+  where
+  eqRaw a b =
+    btbrSpendInputs a == btbrSpendInputs b
+      && btbrCollateralInputs a == btbrCollateralInputs b
+      && btbrReferenceInputs a == btbrReferenceInputs b
+      && btbrOutputs a `eqSeqUnsized` btbrOutputs b
+      && btbrCollateralReturn a `eqMbUnsized` btbrCollateralReturn b
+      && btbrTotalCollateral a == btbrTotalCollateral b
+      && btbrCerts a == btbrCerts b
+      && btbrWithdrawals a == btbrWithdrawals b
+      && btbrTxFee a == btbrTxFee b
+      && btbrValidityInterval a == btbrValidityInterval b
+      && btbrUpdate a == btbrUpdate b
+      && btbrReqSignerHashes a == btbrReqSignerHashes b
+      && btbrMint a == btbrMint b
+      && btbrScriptIntegrityHash a == btbrScriptIntegrityHash b
+      && btbrAuxDataHash a == btbrAuxDataHash b
+      && btbrTxNetworkId a == btbrTxNetworkId b
+    where
+      eqMbUnsized x y = case (x, y) of
+        (SJust a', SJust b') -> a' `eqUnsized` b'
+        (SNothing, SNothing) -> True
+        _ -> False
+      eqSeqUnsized x y = foldl (\acc (x', y') -> acc && x' `eqUnsized` y') True $ StrictSeq.zip x y
```

Comment:
Gotta make sure lengths are the same :wink:, also there is never a case when lazy left fold is better :grin:  
seq:
```suggestion
      eqSeqUnsized x y =
        length x == length y 
          && foldl' (\acc (x', y') -> acc && x' `eqUnsized` y') True $ StrictSeq.zip x y
```